### PR TITLE
Prevent races across prometheus installs

### PR DESCRIPTION
**Pull request by smarterclayton**
The service account secret may take a small amount of time to create

### CODE REVIEW

**Comment by smarterclayton** [comment]
Tagging to reduce flakes in the queue

File: test/extended/prometheus/prometheus.go [hunk 1]
```diff
@@ -319,17 +319,24 @@ func bringUpPrometheusFromTemplate(oc *exutil.CLI) (ns, host, bearerToken string
...
+			if !strings.HasPrefix(secret.Name, "prometheus-") {
+				continue
+			}
+			bearerToken = string(secret.Data[v1.ServiceAccountTokenKey])
+			break
 		}
-		if !strings.HasPrefix(secret.Name, "prometheus-") {
+		if len(bearerToken) == 0 {
+			e2e.Logf("Waiting for prometheus service account secret to show up")
+			time.Sleep(time.Second)
```
Comment (danwinship) [inline test/extended/prometheus/prometheus.go]: I thought utilwait.ExponentialBackoff was preferred for all waiting-for-stuff loops?

File: test/extended/prometheus/prometheus.go [hunk 2]
```diff
@@ -319,17 +319,24 @@ func bringUpPrometheusFromTemplate(oc *exutil.CLI) (ns, host, bearerToken string
 	}
 
 	waitForServiceAccountInNamespace(oc.AdminKubeClient(), "kube-system", "prometheus", 2*time.Minute)
```
Comment (deads2k) [inline test/extended/prometheus/prometheus.go]: If you're waiting for a service account, why wouldn't you also wait for its secret?  Seems like this helper could wait for both in testing.

Comment (smarterclayton) [reply test/extended/prometheus/prometheus.go]: This shouldn't even be using its own method, there's a common helper.  Will clean up in 3.10

**Pull request merged**